opendevin/core/main.py

```diff
@@ -36,6 +36,7 @@ async def main(
     fake_user_response_fn: Optional[Callable[[Optional[State]], str]] = None,
     sandbox: Optional[Sandbox] = None,
     runtime_tools_config: Optional[dict] = None,
+    sid: str | None = None,
```

Comment:
Out of curiosity: why not for None always set a UUID?
This should also "prevent" the AgentController's sid to be "default".
Just in principle, this shouldn't interfere with anything, or?

Reply:
This is a good point, but I wonder if perhaps we can handle that in the future, because currently we are not authenticating in any way with CLI, and I think that means that if we set some random name here, we wouldn't be able to restore the session at will.

Now, currently we aren't restoring CLI sessions anyway, so one can say it's a non-issue, but restoring CLI sessions is actually implemented already, it's just that it's part of PR 2021. I'm splitting that out and submitting it here today and we'll see. (I personally found it insanely useful when working on anything, and specially work with large histories 🫨 , so I will propose it as part of the improvements we are currently doing.)

Reply:
Of course, it was no objection of mine, just technical curiosity. 😁

Reply:
Perhaps we can do more on this in the callers. Like now, some add a pid, some don't (why not?).